user-contracts-mon/internal/parser.go

```diff
@@ -0,0 +1,55 @@
+package monitor
+
+import (
+	"errors"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	env "github.com/hashicorp/go-envparse"
+)
+
+var invalidCfgError = errors.New("Invalid or Missing Fields in configration file")
+
+func readFile(envPath string) (io.Reader, error) {
+	envFile, err := os.Open(envPath)
+	if err != nil {
+		return strings.NewReader(""), err
+	}
+
+	return envFile, nil
+}
+
+func parseMonitor(envFile io.Reader) (Monitor, error) {
```

Comment:
please refactor this to, parseConfig that returns your config struct and have a `NewMonitor` that takes that config struct as a param